internal/kubernetes/operator/deployment/deployment.go

```diff
@@ -515,19 +516,30 @@ func BuildServerlessDeployments(deploymentStore store.AtlasOperatorClusterStore,
 }
 
 func buildServerlessPrivateEndpoints(deploymentStore store.ServerlessPrivateEndpointsLister, projectID, clusterName string) ([]atlasV1.ServerlessPrivateEndpoint, error) {
-	endpoints, err := deploymentStore.ServerlessPrivateEndpoints(projectID, clusterName, &mongodbatlas.ListOptions{ItemsPerPage: MaxItems})
+	endpoints, err := deploymentStore.ServerlessPrivateEndpoints(projectID, clusterName)
 	if err != nil {
 		return nil, err
 	}
 
 	result := make([]atlasV1.ServerlessPrivateEndpoint, 0, len(endpoints))
 
 	for i := range endpoints {
-		result = append(result, atlasV1.ServerlessPrivateEndpoint{
-			Name:                     endpoints[i].Comment,
-			CloudProviderEndpointID:  endpoints[i].CloudProviderEndpointID,
-			PrivateEndpointIPAddress: endpoints[i].PrivateEndpointIPAddress,
-		})
+		endpoint := endpoints[i].GetActualInstance()
+
+		switch v := endpoint.(type) {
+		case *mongodbatlasv2.ServerlessAWSTenantEndpoint:
+			result = append(result, atlasV1.ServerlessPrivateEndpoint{
+				Name:                     *v.Comment,
+				CloudProviderEndpointID:  *v.CloudProviderEndpointId,
+				PrivateEndpointIPAddress: "",
+			})
+		case *mongodbatlasv2.ServerlessAzureTenantEndpoint:
```

Comment:
Q: Do we have sdk issue on this. It looks like we have distinction between objects but they are exactly the same

Reply:
They are not fully the same, for example `PrivateEndpointIPAddress` doesn't exist on the AWS implementation. But here only some of the common properties are used